### DISCUSSION TITLE
0.1.0

### DISCUSSION THREAD

**Discussion by mbrandonw** [pointfreeco/swift-clocks#1] [Announcements]
⏰ A few clocks that make working with Swift concurrency more testable and more versatile.

<hr /><em>This discussion was created from the release <a href='https://github.com/pointfreeco/swift-clocks/releases/tag/0.1.0'>0.1.0</a>.</em>

**Comment by davdroman** [comment]
Loving this release!

One question. Am I right in assuming that dot-syntax shorthands are not available due to Swift's lack of extensions on existentials (for now)?

```swift
#if compiler(>=5.8) // theoretical Swift version where existential extensions are supported
@available(iOS 16, macOS 13, tvOS 16, watchOS 9, *)
extension any Clock<Duration> {
    @inlinable
    public static var immediate: ImmediateClock { .init() }

    @inlinable
    public static var test: TestClock { .init() }

    @inlinable
    public static func unimplemented(
        name: String = "Clock",
        now: Instant = .init()
    ) -> UnimplementedClock {
        .init(name: name, now: now)
    }
}
#endif
```

If so, this could make for a compelling snippet in a potential future proposal.

**Reply by mbrandonw** [reply]
We actually do have support for the dot-syntax shorthands:

https://github.com/pointfreeco/swift-clocks/blob/7379ef30c447ea6060bfa5ae7fab77d0d6e47866/Sources/Clocks/ImmediateClock.swift#L157-L177

This should allow for things like:

```swift
store.dependencies.continuousClock = .immediate
```

But if you try using them it crashes the compiler 😆. We filed a bug for it:

https://github.com/apple/swift/issues/61645

**Reply by davdroman** [reply]
Excellent, thanks for surfacing this.

I knew about this crashing the compiler but I thought it was because it wasn't supposed to be used for existentials. Hopefully it's more feasable to have it fixed than to implement a whole new feature.
